PhysicsTools/Heppy/python/analyzers/objects/TauAnalyzer.py

```diff
@@ -92,6 +91,7 @@ def id6(tau,X):
             elif tau.tauID(self.cfg_ana.tauLooseID):
                 event.looseTaus.append(tau)
                 event.inclusiveTaus.append(tau)
+        event.looseTaus.extend(event.selectedTaus)
 
```

Comment:
Hi Andrea, 

Adding Jan, just in case. 

I understand the following:

- Before the modification:

selectedTaus contains taus that pass the tauID
looseTaus contains taus that pass the tauLooseID
inclusiveTaus contains both

- After the modification:
selectedTaus and inclusiveTaus are unchanged. 
looseTaus contains taus that pass the tauLooseID and the contents of selectedTaus, 
meaning the taus that pass the tauLooseID and the tauID.

It seems to me that it therefore becomes identical to inclusiveTaus, isn’t it? 
If it’s the case, what is the need to have two collections with the same contents? 

In any case, the code is not very readable as it is. 
One should append looseTaus in the loop completely and not extend it after the loop. 

Cheers, 

Colin 

Le 1 avr. 2015 à 17:44, arizzi <notifications@github.com> a écrit :

> In PhysicsTools/Heppy/python/analyzers/objects/TauAnalyzer.py:
> 
> > @@ -92,6 +91,7 @@ def id6(tau,X):
> >              elif tau.tauID(self.cfg_ana.tauLooseID):
> >                  event.looseTaus.append(tau)
> >                  event.inclusiveTaus.append(tau)
> > +        event.looseTaus.extend(event.selectedTaus)
> >  
> @gpetruc @cbernet , hi guys this makes sense to me... why should the loose being exclusive with the selected? if you agree we will make the same PR against colin branch
> 
> —
> Reply to this email directly or view it on GitHub.
>